### PR TITLE
Rename PAL_CMAKE_* CLR_CMAKE_*

### PR DESCRIPTION
**Pull request by sdmaclea**
Remove separatation between CLR_CMAKE and PAL_CMAKE defines

Revise PAL_CMAKE_PLATFORM* defines to use CLR_CMAKE_HOST* defines

Revise other PAL_CMAKE_* defines to use CLR_CMAKE_* equivalents.

### CODE REVIEW

File: src/coreclr/src/pal/src/CMakeLists.txt [hunk 1]
```diff
@@ -68,37 +68,37 @@ if(CLR_CROSS_COMPONENTS_BUILD)
     if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
         if(CLR_CMAKE_TARGET_ARCH STREQUAL "arm" OR CLR_CMAKE_TARGET_ARCH STREQUAL "armel")
             if($ENV{CROSSCOMPILE} STREQUAL "1")
-                set(PAL_CMAKE_PLATFORM_ARCH_I386 1)
+                set(CLR_CMAKE_HOST_ARCH_I386 1)
```
Comment (jkotas) [inline src/coreclr/src/pal/src/CMakeLists.txt]: Is this whole block really needed here ?

If I am reading the cmake files correctly, we should be getting the definitions from configurecompiler.cmake. `add_subdirectory(src/pal)` is after `include(configurecompiler.cmake)` in src\coreclr\CMakeLists.txt.

File: src/coreclr/src/pal/src/CMakeLists.txt [hunk 2]
```diff
@@ -126,24 +126,24 @@ add_definitions(-DFEATURE_PAL)
 add_definitions(-DCORECLR)
 add_definitions(-DPIC)
 add_definitions(-D_FILE_OFFSET_BITS=64)
-if(PAL_CMAKE_PLATFORM_ARCH_AMD64)
+if(CLR_CMAKE_HOST_ARCH_AMD64)
   add_definitions(-DBIT64)
   set(PAL_ARCH_SOURCES_DIR amd64)
```
Comment (hoyosjs) [inline src/coreclr/src/pal/src/CMakeLists.txt]: Configurecompiler also sets these to `ARCH_SOURCES_DIR`

**Review by hoyosjs — COMMENTED** [review]
Actually, checking all these defines are included in configurecompiler.cmake

**Review by hoyosjs — APPROVED** [review]

**Comment by sdmaclea** [comment]
`PAL_ARCH_SOURCES` was set based on `HOST` arch.  Where `ARCH_SOURCES` was set based on `TARGET`.  

When I removed `PAL_ARCH_SOURCES`, it broke becuase of this difference.

I believe the code must not be used for Cross Component Builds, because it is clearly wrong. So I am attempting to simply not add the incorrect code.

**Comment by sdmaclea** [comment]
I restored the odd `PAL_ARCH_SOURCES`.  I couldn't simply not include the code.

These function references were missing
```
DBG_DebugBreak
DBG_DebugBreak_End
ThrowExceptionFromContextInternal
CONTEXT_CaptureContext
RtlCaptureContext
ExecuteHandlerOnOriginalStack
```

It is odd and a bit hacky, but I don't think it is broken, because I doubt any of these functions are actually used in the CROSS_COMPONENTs.

It probably should be fixed eventually.

**Comment by hoyosjs** [comment]
> PAL_ARCH_SOURCES was set based on HOST arch. Where ARCH_SOURCES is set based on TARGET.
> When I removed PAL_ARCH_SOURCES, it broke because of this difference. It was trying to compile arm64 assembly to run on an x64 host.

> I restored the odd `PAL_ARCH_SOURCES`. I couldn't simply not include the code.
> These function references were missing ...
> It is odd and a bit hacky, but I don't think it is broken, because I doubt any of these functions are actually used in the CROSS_COMPONENTs.
> It probably should be fixed eventually.

Interesting 😄. Defines and options never cease to amaze me. As you said, we should fix it, just probably not in this PR. Those are functions that are never really necessary in - sans the context one, which I am not sure off the top of my head. Those are debugger and EH support functions. This still merits a revisit. Still love the build cleanup.

**Pull request merged**